base/single_thread_task_runner.cc

```diff
@@ -47,7 +48,15 @@ void SingleThreadTaskRunner::PostBlockingTask(const base::Location& from_here,
 
   if (task_may_run) {
     // Wait for the task to complete before proceeding.
-    task_finished.Wait();
+    do {
+      if (task_finished.TimedWait(base::TimeDelta::FromMilliseconds(1000))) {
+        break;
+      }
+      base::debug::StackTrace trace;
+      trace.PrintWithPrefix("[task runner deadlock]");
```

Comment:
I think we should put the stacktrace part inside a `#if !defined(COBALT_BUILD_TYPE_GOLD)`.